src/barChart.ts

```diff
@@ -94,31 +99,36 @@ module powerbi.extensibility.visual {
         let barChartDataPoints: BarChartDataPoint[] = [];
         let dataMax: number;
 
-        let colorPalette: IColorPalette = host.colorPalette;
+        let colorPalette: ISandboxExtendedColorPalette = host.colorPalette;
         let objects = dataViews[0].metadata.objects;
+
+        const strokeColor: string = getColumnStrokeColor(colorPalette);
+
         let barChartSettings: BarChartSettings = {
             enableAxis: {
                 show: getValue<boolean>(objects, 'enableAxis', 'show', defaultSettings.enableAxis.show),
+                fill: getAxisTextFillColor(objects, colorPalette, defaultSettings.enableAxis.fill),
             },
             generalView: {
                 opacity: getValue<number>(objects, 'generalView', 'opacity', defaultSettings.generalView.opacity),
                 showHelpLink: getValue<boolean>(objects, 'generalView', 'showHelpLink', defaultSettings.generalView.showHelpLink),
+                helpLinkColor: strokeColor,
             }
         };
+
         for (let i = 0, len = Math.max(category.values.length, dataValue.values.length); i < len; i++) {
-            let defaultColor: Fill = {
-                solid: {
-                    color: colorPalette.getColor(category.values[i] + '').value
-                }
-            };
+            const color: string = getColumnColorByIndex(category, i, colorPalette);
+
+            const selectionId: ISelectionId = host.createSelectionIdBuilder()
+                .withCategory(category, i)
+                .createSelectionId();
 
             barChartDataPoints.push({
-                category: category.values[i] + '',
+                color,
+                strokeColor,
```

Comment:
Please use thicker strokes for high-contrast mode (native visuals use `stroke-width 2`)

Reply:
Done